clients/include/testing_scal.hpp

```diff
@@ -62,6 +62,10 @@ void testing_scal(const Arguments& arg)
         }
 
         CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+        // if(incx <= 0)
+        //     EXPECT_ROCBLAS_STATUS(
+        //         (rocblas_scal<T, U>)(handle, N, &h_alpha, dx, incx), rocblas_status_invalid_size);
+        // else
```

Comment:
`incx <= 0` is not an error for `SCAL`, but just makes it act as a no-op. Rather than doing a separate test and early return, perhaps it should just call `SCAL` normally with `incx <= 0` and then verify that the result has not changed. The case here would be limited to `n <= 0` but `incx <= 0` would fall through to the code which calls CBLAS and compares the results.

Reply:
Removed commented-out code. incx <= 0 still does a separate test and returns early. If this does not happen, incx is used for memory alloc, etc which causes issues.